Add a clock dashboard with a digital clock display for different timezones

nainar opened the pull request:
This patch adds a clock dashboard with a digital clock display for different timezones.

The times are formatted as:
If it is a weekend -> the day is shown in red
If it is working hours in the timezone -> the time is shown green
If it is not working hours in the timezone -> the time is shown red.

Also modifies the blink.json to add the timezones the team cares about.

PTAL?

Thanks!

shans committed:
Add multiple columns and balancing to cardview.
Update redirect to point to monorail.

shans committed:
Fix issues feed.

Commit:
rough draft

Commit:
add time dash board

dstoc commented on client/cz-clock-dash.html:
```diff
@@ -0,0 +1,108 @@
+      },
+      mutateTimeDay: function(timezone, idx) {
+        this.set('timezones.' + idx + '.data', this.currentTimeDay(timezone));
+        this.async(()=>this.mutateTimeDay(timezone, idx), 500);
```
why is this async?

dstoc replied:
I think you should format this as `...async(() => this...`

dstoc replied:
> why is this async?

Oh, this is the loop. 500ms seems a bit fast? Don't we only need to update once a minute? It also appears that when the timezones in the source change we will end up with multiple loops running.

nainar replied:
Done.

dstoc commented on client/cz-clock-dash.html:
```diff
@@ -0,0 +1,108 @@
+        <div class='card-flex'>
+          <template is="dom-repeat" items="{{timezones}}">
+            <paper-item-body two-line>
+              <div id="timer" inner-h-t-m-l=[[item.data]]></div>
```
Why set innerhtml? You should be able to set up the `<span>` using data binding and use a `class` attribute for styling: `<span class="{{item.data.class}}">{{item.data.time}}</span>`

nainar replied:
Done.

dstoc commented on client/cz-clock-dash.html:
```diff
@@ -0,0 +1,108 @@
+      attached: function() {
+        registerSource('cz-config', 'timezones', function(timezones) {
+          this.set('timezones', timezones);
+        }.bind(this));
```
using an arrow function should avoid the need to bind here

nainar replied:
Done.

dstoc commented on client/cz-clock-dash.html:
```diff
@@ -0,0 +1,108 @@
+
+  <script>
+    Polymer({
+      is: "cz-clock-dash",
```
google javascript style [prefers single quotes over double quotes](https://google.github.io/styleguide/javascriptguide.xml?showone=Strings#Strings)

nainar replied:
Done.

Commit:
addressing comments and added timezones of interest for platform team

nainar commented:
@dstockwell, Made the changes you asked for, PTAL?

dstoc commented on configs/platform.json:
```diff
@@ -13,5 +13,12 @@
+"timezones": [
+  {"city": "MTV", "utc": "-7"},
+  {"city": "NYC", "utc": "-4"},
+  {"city": "LON", "utc": "0"},
```
London and Munich are the same number of characters as Sydney.

dstoc commented on client/cz-clock-dash.html:
```diff
@@ -0,0 +1,111 @@
+      },
+      mutateTimeString: function() {
+        this.timezones.forEach((timezone, idx) => this.set('timezones.' + idx + '.data', this.currentTimeDay(timezone)));
+        this.async(()=>this.mutateTimeString(), 1000);
```
spaces

dstoc commented:
lgtm

Commit:
switch to 24 hour clock, remove seconds and add spaces

Commit:
timezone format errors

nainar commented:
Changes made since last lgtm:
1. Changed timezone formatting in platform.json
2. Switched to 24 hour clock
3. Removed seconds
4. Added space -> formatting error.

dstoc commented:
For future reference, here's an API to go from lat+long to time offset: https://maps.googleapis.com/maps/api/timezone/json?location=-33.8650,151.2094&timestamp=1331161200

shans committed:
review tweaks

shans committed:
Merge pull request #8 from shans/multicol

Add multiple columns and balancing to cardview.

Commit:
merge 1
